util/dvsim/Testplan.py

```diff
@@ -213,15 +223,25 @@ def _create_testplan_elements(kind, raw_dicts_list):
                 print(f"Error: {kind} arg is invalid.\n{e}")
                 sys.exit(1)
             except ValueError as e:
-                print(e)
+                print(f"{kind}\n{dict_entry}\n{e}")
                 sys.exit(1)
 
             if item.name in item_names:
                 print(f"Error: Duplicate {kind} item found with name: "
                       f"{item.name}")
                 sys.exit(1)
-            items.append(item)
-            item_names.add(item.name)
+
+            # If tags are provided, then filter the testpoints by the provided
+            # tags.
+            if tags:
+                valid = hasattr(item, "tags") and set(tags).issubset(
+                    set(item.tags))
+            else:
+                valid = True
```

Comment:
I'd suggest defining a `tags_are_compatible()` function to handle this? (I think the test is "Either the item has no tags or it has every tag in `tags`"? Probably worth writing out explicitly?)

(Also, the hasattr stuff goes away if we set a default value like I suggested above)